src/app/public/modules/modal-provider/modal-provider.service.ts

```diff
@@ -0,0 +1,26 @@
+import {
+  Inject,
+  Injectable,
+  Optional
+} from '@angular/core';
+
+import {
+  SkyModalProvider
+} from './types/modal-provider';
+
+import {
+  SKY_MODAL_PROVIDER
+} from './types/modal-provider-injection-token';
+
+@Injectable()
+export class SkyModalProviderService {
+
+  constructor(
+    @Inject(SKY_MODAL_PROVIDER) @Optional() private modalProviders?: SkyModalProvider[]
+  ) {
+  }
+
+  public getModalForType(type: string): SkyModalProvider {
+    return this.modalProviders?.find(provider => provider.type === type);
```

Comment:
Will there ever be a need to return multiple providers of a specific type? This will return the first provider of a specific type.

Reply:
That seems problematic. We then wouldn't know which provider to use. We can discuss in standup but I think we should just have one for each.